File: scripts/fix-page.js
const fs = require('fs');
const filePath = 'C:\\website\\hanyangpetro-ops\\src\\app\\admin\\orders\\[id]\\page.tsx';
let text = fs.readFileSync(filePath, 'utf8');

const old1 = "const canStartHanwhaOrder = session.user.name === '\uC591\uD76C\uCCA0';";
const new1 = "const canStartHanwhaOrder = session.user.userKind === 'staff';";
console.log('old1 found:', text.includes(old1));
text = text.split(old1).join(new1);

const old2 = "const isInternalPurchaseOnly = normalizeCompanyName(order.customer.companyName) === '\uD55C\uC591\uC720\uD654';";
const addition = "\n    const hasNonHanwhaSupplier = order.items.some((item) => {\n        const name = item.purchaseSupplier?.supplierName ?? '';\n        return name !== '' && !name.includes('\uD55C\uD654');\n    });\n    const isDispatchWaiting = order.status === 'DISPATCH_WAITING';";
console.log('old2 found:', text.includes(old2));
text = text.split(old2).join(old2 + addition);

fs.writeFileSync(filePath, text, 'utf8');
console.log('done');

File: scripts/fix-page.py
path = r'C:\website\hanyangpetro-ops\src\app\admin\orders\[id]\page.tsx'
with open(path, 'rb') as f:
    data = f.read()
text = data.decode('utf-8')

# 1. canStartHanwhaOrder: 모든 직원으로 변경
old1 = "const canStartHanwhaOrder = session.user.name === '양희철';"
new1 = "const canStartHanwhaOrder = session.user.userKind === 'staff';"
count1 = text.count(old1)
text = text.replace(old1, new1)
print(f'1. replaced {count1} occurrences of canStartHanwhaOrder')

# 2. isInternalPurchaseOnly 이후에 새 변수 추가
old2 = "const isInternalPurchaseOnly = normalizeCompanyName(order.customer.companyName) === '한양유화';"
new2 = """const isInternalPurchaseOnly = normalizeCompanyName(order.customer.companyName) === '한양유화';
    // 수기배차 폼: 매입처 중 한화솔루션이 아닌 것이 포함된 경우에만 표시
    const hasNonHanwhaSupplier = order.items.some((item) => {
        const name = item.purchaseSupplier?.supplierName ?? '';
        return name !== '' && !name.includes('한화');
    });
    // 미배차/백오더: DISPATCH_WAITING 상태일 때만
    const isDispatchWaiting = order.status === 'DISPATCH_WAITING';"""
count2 = text.count(old2)
text = text.replace(old2, new2)
print(f'2. replaced {count2} occurrences of isInternalPurchaseOnly')

with open(path, 'wb') as f:
    f.write(text.encode('utf-8'))
print('done')
print('userKind in text:', 'userKind' in text)
print('hasNonHanwhaSupplier in text:', 'hasNonHanwhaSupplier' in text)
print('isDispatchWaiting in text:', 'isDispatchWaiting' in text)
